drawBot/drawBotDrawingTools.py

```diff
@@ -465,6 +465,16 @@ def save(self):
         self._requiresNewFirstPage = True
         self._addInstruction("save")
 
+        class SaveContextManager(object):
+
+            def __enter__(saveCxt):
+                return saveCxt
+
+            def __exit__(saveCxt, type, value, traceback):
+                self.restore()
+
+        return SaveContextManager()
+
```

Comment:
I would write it like so:
```python
# at global level
class SaveContextManager(object):
    def __init__(self, drawingTools):
        self._drawingTools = drawingTools
    def __enter__(self):
        return self
    def __exit__(self type, value, traceback):
        self._drawingTools.restore()

# in save():
    return SaveContextManager(self)
```

Reply:
I didnt want to keep a reference of a callback or an object somewhere...
this feels very similar as a decorator wrapper

Reply:
Why not? To not have `self` as a first argument is very ugly. It's an explicit reference as opposed to a scoped variable, which I think is rather obscure when using a nested class.